### PR TITLE
Release notes for oc adm prune deployments --replica-sets flag

### PR DESCRIPTION
**Pull request by bergerhoffer**
https://issues.redhat.com/browse/OSDOCS-2892

Previews:
* https://deploy-preview-41942--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-10-release-notes#ocp-4-10-dev-exp-prune-deployments
* https://deploy-preview-41942--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-10-release-notes#ocp-4-10-technology-preview

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@deejross @zhouying7780 Can you please review?

Preview: https://deploy-preview-41942--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-10-release-notes#ocp-4-10-dev-exp-prune-deployments

File: release_notes/ocp-4-10-release-notes.adoc [hunk 1]
```diff
@@ -341,6 +341,13 @@ The Special Resource Operator (SRO) now exposes metrics to help you watch the he
 [id="ocp-4-10-dev-exp"]
 === Developer experience
 
+[id="ocp-4-10-dev-exp-prune-deployments"]
+==== Pruning deployment replica sets (Technology Preview)
```
Comment (bergerhoffer) [inline release_notes/ocp-4-10-release-notes.adoc:345]: @soltysh I chatted with @deejross about the flag being experimental, so marking it as TP in the docs. Are you okay with this?

**Comment by ahardin-rh** [comment]
LGTM!

**Comment by bergerhoffer** [comment]
Note to self, if we keep the heading change in #41941, make that adjustment in the link here ("Pruning DeploymentConfig objects" -> "Pruning deployment resources")

File: release_notes/ocp-4-10-release-notes.adoc (hunk 1 above)
Comment (bergerhoffer) [reply release_notes/ocp-4-10-release-notes.adoc:345]: Per slack convo, merging this PR. And if there are any concerns with this being marked TP, we can adjust in a future PR

**Pull request merged**